packages/ocular/src/services/organisation.ts

```diff
@@ -171,33 +170,35 @@ class OrganisationService extends TransactionBaseService {
                 installed_apps[webConnector_index].links = [];
               }
 
-              const linkExist = installed_apps[webConnector_index].links.find(
-                (ele) => ele.id === link_id
-              );
+              const linkExist = installed_apps[
+                webConnector_index
+              ].links.findIndex((ele) => ele.id === data.link_id);
               if (linkExist !== -1) {
                 installed_apps[webConnector_index].links[linkExist] = {
-                  id: link_id,
-                  location: link,
+                  ...installed_apps[webConnector_index].links[linkExist],
                   status: data.status,
                 };
               } else {
                 installed_apps[webConnector_index].links.push({
-                  id: link_id,
-                  location: link,
+                  id: data.link_id,
+                  location: data.link,
                   status: data.status,
+                  title: data.title,
+                  description: data.title,
                 });
               }
 
-              await this.update(this.loggedInUser_.organisation_id, {
+              await this.update(data.org_id, {
                 installed_apps,
               });
             }
             if (data.emit_event) {
               await this.eventBusService_.emit("webConnectorInstalled", {
                 organisation: this.loggedInUser_.organisation,
```

Comment:
What happens if the data.emit_event is true but the app is not a web connector?

Reply:
This emit evet will only be passed as variable in case of Web Connector  !